Degrade a "Pin not found:" Error into a PINNOTFOUND warning so it can be disabled if required

udif opened the pull request:
This small PR implements the suggestion at https://github.com/verilator/verilator/issues/2866#issuecomment-811057315 .
I added a test for the regression, but I'm not sure this is the correct way (not familiar with driver.pl).
I have ran the new test manually and it behaves as expected.

wsnyder commented on src/V3Error.h:
```diff
@@ -131,6 +131,7 @@ class V3ErrorCode final {
         VARHIDDEN,      // Hiding variable
         WIDTH,          // Width mismatch
         WIDTHCONCAT,    // Unsized numbers/parameters in concatenations
+        PINNOTFOUND,    // A pin name in a module instance does not exist in the module being instanced
```
Please insert in sorted order with other messages.  Slightly shorter comment so fits in 90 chars.

Add to pretendError as usually this will be an error.

wsnyder reviewed (commented):
Great, that was quick, thanks.

Please add a description of the error in bin/verilator (sorted into other warnings).

There should be a test that fails because of the new warning text, you'll see that soon once the tests finish.  To fix the failing test run e.g. "test_regress/t/t_lint_pindup_bad.pl --gold"

See below also:

udif commented:
Thanks for the feedback. In my earlier PR I failed to notice that the warnings were sorted alphabetically. While fixing this, I also reordered PINCONNECTEMPTY which was also out of order.
I also fixed a small typo in the test names, and added a 'bad' test.
One thing I don't know how to solve is that the CI scripts insists I have to add 'udif', my github username, into docs/CONTRIBUTORS instead of my full name , Udi Finkelstein.

wsnyder commented on bin/verilator:
```diff
@@ -4733,6 +4733,36 @@ Warns that an instance has a pin which is not connected to another signal.
+instanciated. Please note that Verilator raises these errors also on instances
+that should be disabled by generate/if/endgenerate constructs:
+
+    module a;
+    localparam A=1;
+    generate
+    if (A==0)
+    begin
+    b b_inst1 (.x(1'b0)); // nonexistent port
+    b #(.PX(1'b0)) b_inst2 (); // nonexistent parameter
+    end
+    endgenerate
+    endmodule
```
```suggestion
    module a;
      localparam A=1;
      generate
         if (A==0) begin
            b b_inst1 (.x(1'b0)); // nonexistent port
            b #(.PX(1'b0)) b_inst2 (); // nonexistent parameter
         end
       endgenerate
    endmodule
```

wsnyder commented on bin/verilator:
```diff
@@ -4733,6 +4733,36 @@ Warns that an instance has a pin which is not connected to another signal.
+    module b;
+    endmodule
+
+In the example above, b is instanciated with a port named x, but module b has
+no such port at all. In the next line, b is instantiated again, this tim with
+a nonexistent parameter PX. Technically, this code is in error because of
+this, but other commercial tools ignore it because the generate/if condition
+is false, and so module b in this example is not instanciated.
```
```suggestion
In the example above, b is instantiated with a port named x, but module b has
no such port. In the next line, b is instantiated again with
a nonexistent parameter PX. Technically, this code is incorrect because of
this, but other may tools ignore it because module b is not instantiated
due to the generate/if condition being false.
```

wsnyder commented on bin/verilator:
```diff
@@ -4733,6 +4733,36 @@ Warns that an instance has a pin which is not connected to another signal.
+
+It is possible to disable the error above using:
+/* verilator lint_off PINNOTFOUND */
+This way it is possible to lint legacy code that other commercial tools accept.
```
```suggestion
```

udif replied:
OK, Fixed my name in the github settings.

wsnyder reviewed (commented):
When I hover over your icon in this message board (to the left of these messages on github.com) it says "udif", you need to update that in your github.com name setting (not your .git/config file).

Few other minor things, then once these fixed and passes plan to merge.

udif commented:
> When I hover over your icon in this message board (to the left of these messages on github.com) it says "udif", you need to update that in your github.com name setting (not your .git/config file).
> 
> Few other minor things, then once these fixed and passes plan to merge.

OK, added my name via my github settings.
